packages/blocks/src/api/registration.js

```diff
@@ -723,6 +723,10 @@ export const getBlockVariations = ( blockName, scope ) => {
  * ```
  */
 export const registerBlockVariation = ( blockName, variation ) => {
+	if ( typeof variation.name !== 'string' ) {
+		console.warn( 'Variation names must be unique strings.' );
+	}
```

Comment:
This could use a simple unit test. We're already testing all the `console.error` instances in `registerBlockType()` for example.

Reply:
Added in [238d4d0](https://github.com/WordPress/gutenberg/pull/61037/commits/238d4d0fb763e769b3d5be6e034894a57e2247b2)

Reply:
Thanks! Feel free to 🚢 when green.